src/weekly/mod.rs

```diff
@@ -51,6 +51,71 @@ impl WeeklyCalendar {
         self.windows.query_point(timepoint).count() > 0
     }
 
+    /// Return the duration remaining till the next window containing the given datetime.
+    ///
+    /// This returns `None` if no windows are reachable.
+    pub fn remaining_to_datetime(&self, datetime: &DateTime<Utc>) -> Option<chrono::Duration> {
+        if self.is_empty() {
+            return None;
+        }
+
+        // Already in a window, zero minutes.
+        if self.contains_datetime(&datetime) {
+            return Some(chrono::Duration::zero());
+        }
+
+        let timepoint = utils::datetime_as_weekly_minute(datetime);
+        // Next window is this week, just subtract remaining minutes.
+        if let Some(next) = self
+            .windows
+            .iter_sorted()
+            .find(|x| x.range.start >= timepoint)
+        {
+            let remaining_mins = next.range.start.saturating_sub(timepoint);
+            return Some(chrono::Duration::minutes(i64::from(remaining_mins)));
+        };
+
+        // Next window is not this week, wrap remaining minutes to the first
+        // window of the next week (calendar has been already verified non-empty).
+        let remaining_mins = {
+            let remaining_this_week: i64 = MAX_WEEKLY_MINS.saturating_sub(timepoint).into();
+            let first_window_next_week = self
+                .windows
+                .iter_sorted()
+                .next()
+                .expect("unexpected empty weekly calendar");
+            remaining_this_week.saturating_add(first_window_next_week.range.start.into())
+        };
+        Some(chrono::Duration::minutes(remaining_mins))
+    }
+
+    /// Format remaining duration till the next window in human terms.
+    pub fn human_remaining_duration(remaining: &chrono::Duration) -> String {
+        let mut human = String::new();
+        if remaining.is_zero() {
+            human.push_str("now");
+            return human;
+        }
+
+        let earlier_output;
+        human.push_str("in ");
+        let days = remaining.num_days() % 7;
+        if days > 0 {
+            human.push_str(&format!("{}d ", days));
+            earlier_output = true;
+        } else {
+            earlier_output = false;
+        }
+        let hours = remaining.num_hours() % 24;
+        if hours > 0 || earlier_output {
+            human.push_str(&format!("{}h ", hours));
```

Comment:
Not that it really matters here but I think it's better to do
`write!(&mut human, "{}h", hours);`
to avoid box allocating an intermediate string.